engine/tests/ElggCoreSearchTest.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+/**
+ * Test elgg_search()
+ */
+class ElggCoreSearchTest extends \ElggCoreGetEntitiesBaseTest {
+
+	public function testCanSearchUsersByAttributeFields() {
+
+		$users = elgg_search([
+			'type' => 'user',
+			'fields' => ['username'],
+			'query' => 'test_user_',
+		]);
+
+		$this->assertTrue((bool) $users);
+		foreach ($users as $user) {
+			$this->assertEqual(1, preg_match('/test_user_/i', $user->username));
+		}
+	}
+
+	public function testCanSearchUsersByExactMatchInMetadataFields() {
+
+		$callback = function(\Elgg\Hook $hook) {
+			$fields = $hook->getValue();
+			$fields[] = 'country';
+			return $fields;
+		};
+
+		elgg_register_plugin_hook_handler('search:fields', 'user', $callback);
+
+		$users = elgg_search([
+			'type' => 'user',
+			'fields' => ['country'],
+			'query' => 'united states',
+			'partial_match' => false,
+			'tokenize' => false,
+		]);
+
+		$this->assertTrue((bool) $users);
+		foreach ($users as $user) {
+			$this->assertEqual('United States', $user->country);
+		}
+
+		elgg_unregister_plugin_hook_handler('search:fields', 'user', $callback);
+	}
+
+	public function testCanSearchUsersByTokensInMetadataFields() {
+
+		$callback = function(\Elgg\Hook $hook) {
+			$fields = $hook->getValue();
+			$fields[] = 'country';
+			return $fields;
+		};
+
+		elgg_register_plugin_hook_handler('search:fields', 'user', $callback);
+
+		$users = elgg_search([
+			'type' => 'user',
+			'fields' => ['country'],
+			'query' => ' argentina ',
```

Comment:
Not sure what tokenization without a partial match should do... It kinds of doesn't make sense to tokenize something if you are not allowing partial matches.